tests/test_main.py

```diff
@@ -1,6 +1,182 @@
-# TODO: add tests
-from app.main import hello_world
+import ast
+import inspect
+import io
+from contextlib import redirect_stdout
 
+import pytest
 
-def test_hello_world():
-    assert hello_world() == "Hello, world!"
+import app.main as main
+
+
+@pytest.mark.parametrize("class_name", ["Animal", "Cat", "Dog"])
+def test_classes_should_be_defined(class_name):
+    assert hasattr(main, class_name)
+
+
+@pytest.mark.parametrize(
+    "class_name,methods",
+    [
+        ("Animal", ["print_name", "feed"]),
+        ("Cat", ["print_name", "feed", "catch_mouse"]),
+        ("Dog", ["print_name", "feed", "bring_slippers"]),
+    ],
+)
+def test_classes_should_have_corresponding_methods(class_name, methods):
+    class_to_test = getattr(main, class_name)
+    for method in methods:
+        assert (
+            hasattr(class_to_test, method) is True
+        ), f"Class '{class_to_test}' should have method {method}"
+
+
+@pytest.mark.parametrize(
+    "class_name,method",
+    [
+        ("Cat", "catch_mouse"),
+        ("Dog", "bring_slippers"),
+    ],
+)
+def test_only_one_method_should_be_declared_in_each_of_children_classes(
```

Comment:
Good job!
You've automated even this check with tests!
(In GitHub tasks sometimes we can check just with eyes, but this is really good to check it this way :)

Reply:
I just copied it with new params from code editor task :)
From pytest.mark.parametrize with love)